pkg/endpoint/endpoint.go

```diff
@@ -2422,8 +2426,24 @@ func (e *Endpoint) syncPolicyMap() error {
 
 	currentMapContents, err := e.PolicyMap.DumpToSlice()
```

Comment:
Would it be possible to move some of this dump/close/retry logic to a helper function? That would allow some context in the doc comment, and make it clear that all this logic is grouped together.